onnxruntime/python/tools/transformers/onnx_exporter.py

```diff
@@ -375,46 +401,88 @@ def export_onnx_model_from_tf(model_name, opset_version, use_external_data_forma
+                                           padding="max_length",
                                            truncation=True)
-
     example_inputs = filter_inputs(example_inputs, input_names)
 
-    example_outputs = model(example_inputs, training=False).to_tuple()
-
-    # Flatten is needed for gpt2 and distilgpt2.
-    example_outputs_flatten = flatten(example_outputs)
-    example_outputs_flatten = update_flatten_list(example_outputs_flatten, [])
+    if config.is_encoder_decoder:
+        example_inputs["decoder_input_ids"] = tokenizer.encode_plus("This is a sample input",
+                                                                    return_tensors="tf",
+                                                                    max_length=max_input_size,
+                                                                    padding="max_length",
+                                                                    truncation=True).input_ids
+    if model_name == "unc-nlp/lxmert-base-uncased":
+        example_inputs["visual_feats"] = tf.random.normal([1, 1, config.visual_feat_dim])
+        example_inputs["visual_pos"] = tf.random.normal([1, 1, config.visual_pos_dim])
+
+    try:
+        # Use no past state for these models
+        if config.use_cache:
+            config.use_cache = False
+    except:
+        pass
+
+    example_outputs = model(example_inputs, training=False)
+    output_names = None
+    
+    # For xlnet models, only compare the last_hidden_state output. 
+    if model_name == "xlnet-base-cased" or model_name == "xlnet-large-cased":
+        output_names = ["last_hidden_state"]
+        example_outputs = example_outputs["last_hidden_state"]
+
+    # Flatten is needed for gpt2 and distilgpt2. Output name sorting is needed for tf2onnx outputs to match onnx outputs. 
+    from tensorflow.python.util import nest
+    example_outputs_flatten = nest.flatten(example_outputs)
 
     onnx_model_path = get_onnx_file_path(onnx_dir, model_name, len(input_names), False, use_gpu, precision, False,
                                          use_external_data_format)
+    tf_internal_model_path = onnx_model_path[:-5] if use_external_data_format else onnx_model_path
 
-    if overwrite or not os.path.exists(onnx_model_path):
+    if overwrite or not os.path.exists(tf_internal_model_path):
         logger.info("Exporting ONNX model to {}".format(onnx_model_path))
-        Path(onnx_model_path).parent.mkdir(parents=True, exist_ok=True)
+        if not use_external_data_format:
+            Path(tf_internal_model_path).parent.mkdir(parents=True, exist_ok=True)
+
+        import tf2onnx, zipfile
+        tf2onnx.logging.set_level(tf2onnx.logging.ERROR)
+        specs = []
+        for name, value in example_inputs.items():
+            dims = [None] * len(value.shape)
+            specs.append(tf.TensorSpec(tuple(dims), value.dtype, name=name))
+        _, _ = tf2onnx.convert.from_keras(model,
```

Comment:
does tf2onnx support dynamic axis right now?

Reply:
Yes it does!